TooltipDecorator/useTooltip.js

```diff
@@ -0,0 +1,266 @@
+/* global MutationObserver ResizeObserver */
+
+import {forProp, forward, handle, not} from '@enact/core/handle';
+import {Job} from '@enact/core/util';
+import {useI18nContext} from '@enact/i18n/I18nDecorator';
+import {FloatingLayerBase} from '@enact/ui/FloatingLayer';
+import ri from '@enact/ui/resolution';
+
+import {Tooltip, defaultArrowAnchor, defaultDirection} from './Tooltip';
+import {adjustDirection, adjustAnchor, calcOverflow, getLabelOffset, getPosition} from './util';
+
+let currentTooltip; // needed to know whether or not we should stop a showing job when unmounting
+
+import useHandlers from '@enact/core/useHandlers';
+import {useCallback, useEffect, useLayoutEffect, useRef, useState} from 'react';
+
+function getDirectionAnchor (tooltipPosition, tooltipType) {
+	const position = tooltipPosition || (defaultDirection(tooltipType) + ' ' + defaultArrowAnchor(tooltipType));
+	const arr = position.split(' ');
+
+	if (arr.length === 2) {
+		return {
+			tooltipDirection: arr[0],
+			arrowAnchor: arr[1]
+		};
+	} else {
+		return {
+			tooltipDirection: position === 'below' ? 'below' : 'above',
+			arrowAnchor: 'right'
+		};
+	}
+}
+
+const removeTooltipProps = ({...props}) => {
+	delete props.rtl;
+	delete props.screenEdgeKeepout;
+	delete props.tooltipDelay;
+	delete props.tooltipMarquee;
+	delete props.tooltipPosition;
+	delete props.tooltipProps;
+	delete props.tooltipRelative;
+	delete props.tooltipText;
+	delete props.tooltipType;
+	delete props.tooltipUpdateDelay;
+	delete props.tooltipWidth;
+
+	return props;
+};
+
+// A hook to show Sandstone-styled tooltip components.
+function useTooltip (props = {}) {
+	const {
```

Comment:
Would you please add propTypes for the props?

Reply:
Let's define these props one in a line.

Reply:
This is a hook, so propTypes not working. I will update props to be one in a line.